Illumio/lib/illumio/infrastructure/containercluster.py

```diff
@@ -0,0 +1,184 @@
+        ...     container_workload_profile, parent=container_cluster
+        ... )
+        >>> container_workload_profile
+        ContainerWorkloadProfile(
+            href='/orgs/1/container_clusters/f5bef182-8c55-4219-b35b-0a50b707e434/container_workload_profiles/d2d466b5-106d-48e9-ada9-68f6321d1da8',
+            name='illumio-system',
+            namespace=None,
+            managed=True,
+            labels=[
+                LabelRestriction(
+                    key='env',
+                    assignment=Reference(href='/orgs/1/labels/23'),
+                    ...
+                ),
+                ...
+            ],
+            enforcement_mode='visibility_only'
+            ...
+        )
+    """
+    namespace: str = None
+    assign_labels: List[Reference] = None
+    labels: List[LabelRestriction] = None
+    enforcement_mode: str = None
+    visibility_level: str = None
+    linked: bool = None
+    managed: bool = None
+
+    def _validate(self):
+        if self.enforcement_mode and self.enforcement_mode not in EnforcementMode:
+            raise IllumioException("Invalid enforcement_mode: {}".format(self.enforcement_mode))
+        if self.visibility_level and self.visibility_level not in VisibilityLevel:
+            raise IllumioException("Invalid visibility_level: {}".format(self.visibility_level))
+        return super()._validate()
+
+
+@dataclass
+@pce_api('container_clusters')
+class ContainerCluster(IllumioObject):
+    """Represents a container cluster object in the PCE.
+
+    Container clusters are abstract representations of container orchestration
+    systems linked to the PCE.
+
+    **NOTE:** when a container cluster is created through the API, the
+    `container_cluster_token` used by Kubelink and C-VEN containers
+    to pair with the PCE is returned in the response. This token is
+    only available after the initial POST request and cannot be
+    retrieved via the API: make sure to store it in a persistent
+    form after creating the cluster.
+
+    See https://docs.illumio.com/core/21.5/Content/LandingPages/Guides/kubernetes-and-openshift.htm
+
+    Usage:
+        >>> import illumio
+        >>> pce = illumio.PolicyComputeEngine('pce.company.com', port=443, org_id=1)
+        >>> pce.set_credentials('api_key', 'api_secret')
+        >>> container_cluster = illumio.ContainerCluster(
+        ...     name='CC-EKS-PROD',
+        ...     description='Production Kubernetes cluster on AWS'
+        ... )
+        >>> container_cluster = pce.container_clusters.create(container_cluster)
+        >>> container_cluster
+        ContainerCluster(
+            href='/orgs/1/container_clusters/f5bef182-8c55-4219-b35b-0a50b707e434',
+            name='CC-EKS-PROD',
+            description='Production Kubernetes cluster on AWS',
+            container_cluster_token='1_016dace1ab35fafe8e71c6dda6695e0881393f1f4c494e6cd70178f1e743b372',
```

Comment:
:exclamation:**Cycode**: Secret of type: 'Generic Password' was found.
Severity: Critical

Would you like to exclude this secret from your status checks?
Tell us what to do with one of the following hashtags:

| Tag | Short Description |
| --- | --- |
| **#cycode_secret_ignore_here** | Applies to this request only |
| **#cycode_secret_false_positive** | Applies to this secret value for all repos in your organization |

 <!-- 23b28f49-b02b-4921-bb8f-0825237109ff -->

Reply:
This is from the Illumio SDK. Resolving it.